src/components/common/datePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,119 @@
+import { Button } from 'components/common/button';
+import { Calendar, CalendarProps } from 'components/common/calendar';
+import { DropdownMenu, MenuButtonProps } from 'components/common/dropdownMenu';
+import { subDays, getUnixTime, isSameDay } from 'date-fns';
+import { memo, ReactNode, useState } from 'react';
+import { DateRange } from 'react-day-picker';
+
+export type DatePickerPreset = {
+  label: string;
+  days: number;
+};
+
+interface Props {
+  button: ReactNode;
+  defaultStart: number;
+  defaultEnd: number;
+  onConfirm: (props: { start: string; end: string }) => void;
+  presets?: DatePickerPreset[];
+  options?: Omit<CalendarProps, 'mode' | 'selected' | 'onSelect'>;
+}
+
+export const DateRangePicker = memo((props: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const Trigger = (attr: MenuButtonProps) => (
+    <button
+      {...attr}
+      onClick={(e) => {
+        setIsOpen(true);
+        attr.onClick(e);
+      }}
+      aria-label="Pick date range"
+      className="flex items-center gap-8"
+    >
+      {props.button}
+    </button>
+  );
+
+  return (
+    <DropdownMenu
+      isOpen={isOpen}
+      setIsOpen={setIsOpen}
+      placement="bottom-start"
+      aria-expanded={isOpen}
+      button={Trigger}
+    >
+      <Content {...props} />
+    </DropdownMenu>
+  );
+});
+
+const Content = (props: Props) => {
+  const [date, setDate] = useState<DateRange | undefined>({
+    from: new Date(Number(props.defaultStart) * 1000),
+    to: new Date(Number(props.defaultEnd) * 1000),
+  });
+  const hasDates = !!(date?.from && date?.to);
+  const selectedPreset = props.presets?.find((p) => {
+    if (!hasDates) {
+      return false;
+    }
+
+    const from = subDays(new Date(), p.days);
+    return isSameDay(from, date?.from!);
+  });
+
+  const handlePreset = (days: number) => {
+    setDate({
+      from: subDays(new Date(), days),
+      to: new Date(),
+    });
+  };
+
+  const onConfirm = () => {
+    if (!hasDates) {
+      return;
+    }
+
+    props.onConfirm({
+      start: getUnixTime(date.from!).toString(),
+      end: getUnixTime(date.to!).toString(),
+    });
+  };
+
+  return (
+    <div className="space-y-20 p-20">
```

Comment:
It does look like a form to me. I would change to :
```html
<!-- You'll need to e.preventDefault in onConfirm -->
<form onSubmit={onConfirm} >
```


Also for the style I would use grid to reduce the amount of divs: 
```html
<div className="grid grid-cols-[200px_1fr] gap-x-30 gap-y-20">
```
Like that you don't need the two div below : 
```html
<div className="flex flex-col gap-5">
  {props.presets.map(...)
</div>
```
And the button will be :
```html
<Button type="submit" disabled={...} className="col-span-2 justify-self-end" size="sm">
```